### ISSUE TITLE
"Editor-Like" AssetSaver example

### ISSUE THREAD

**Issue by viridia** [bevyengine/bevy#14950]
## What problem does this solve or what need does it fill?

Currently there are no examples of how to write an editor that saves assets. I'm not talking about the editor UI, but simply the mechanism for writing assets to disk. The code for saving assets is quite complex and could benefit from some examples.

## What solution would you like?

I propose a very simple editor that only saves one kind of asset. Perhaps a "palette" editor, which lets you save a list of colors, serialized into some format. Or even simpler, a saver that saves a counter (ah, the ubiquitous counter example lives on!) To keep the example small, we would want to avoid anything that requires a complex UI for the editor.

**Comment by djeedai** [comment]
I don't think this is ready for implementation. There's multiple issues about `AssetSaver` being unusable.
